src/constraints/generator_constraints.jl

```diff
@@ -42,21 +42,39 @@ end
 
 function add_binGenIsOnInTS_constraints(m,p)
 	# Generator must be on for nonnegative output
-	@constraint(m, [t in p.techs.gen, ts in p.time_steps],
-		m[:dvRatedProduction][t, ts] <= p.s.generator.max_kw * m[:binGenIsOnInTS][t, ts]
-	)
-	# Note: min_turn_down_pct is only enforced when off_grid_flag is true and in p.time_steps_with_grid, but not for grid outages for on-grid analyses
-	if p.s.settings.off_grid_flag 
-		@constraint(m, [t in p.techs.gen, ts in p.time_steps_without_grid],
-			p.s.generator.min_turn_down_pct * m[:dvSize][t] - m[:dvRatedProduction][t, ts] <=
-			p.s.generator.max_kw * (1 - m[:binGenIsOnInTS][t, ts])
+	if hasproperty(p, Symbol("max_sizes")) #regular REopt model, use p.max_sizes because it includes existing_kw
```

Comment:
It seems like we could simplify all these conditions by using `p.s.generator.max_kw`. We are using `p.s.generator.min_turn_down_pct` and we do not allow for more than one value in `p.techs.gen` so there seems to be no reason to have `t in p.techs.gen` in these constraints. Then we could just apply these constraints to `t="Generator"`. What do you think @hdunham ?

Reply:
I have updated these constraints locally to just apply to "Generator". I have also simplified the code to first check if p is of type REoptInputs. If it is, then set a local variable gen_size to existing kw + max_kw. if it is not REoptInputs type then assume it is MPC type.

Reply:
Also, I think it's good practice to write t in p.techs.gen, even if currently there can only be one generator. Makes the code more robust to future changes.

Reply:
addressed in fb25e34